lib/wallet.py

```diff
@@ -1421,6 +1421,7 @@ def delete_address(value):
         requests = {addr.to_storage_string() : delete_address(value.copy())
                     for addr, value in self.receive_requests.items()}
         self.storage.put('payment_requests', requests)
+        self.storage.write()
```

Comment:
I seem to recall getting bugs related to this on iOS.  I actually had to explicitly call `wallet.storage.write()` on iOS to deal with this issue.  At the time I wanted iOS to be as "non-intrusive" as possible so I didn't bother to issue a PR to fix this. I figured the devs that wrote this knew what they were doing.

Good on you for adding this. it always bugged me. :)